### PR TITLE
fix redundant args of lambda and remove exception of destructor

### PR DESCRIPTION
**Pull request by velconia**

### CODE REVIEW

File: paddle/fluid/operators/distributed/proto_encoder_helper.h [hunk 1]
```diff
@@ -83,7 +83,9 @@ class ProtoEncodeHelper {
 
   ~ProtoEncodeHelper() {
     // Make sure callers didn't do operations that went over max_size promised
-    PADDLE_ENFORCE_LE(p_, limit_);
+#define REPLACE_ENFORCE_GLOG 1
```
Comment (typhoonzero) [inline paddle/fluid/operators/distributed/proto_encoder_helper.h]: Why?

Comment (velconia) [reply paddle/fluid/operators/distributed/proto_encoder_helper.h]: it's ```#define``` here, because we should force to not throw exception here

Comment (typhoonzero) [reply paddle/fluid/operators/distributed/proto_encoder_helper.h]: How about put `throw_on_error` directly.

Comment (velconia) [reply paddle/fluid/operators/distributed/proto_encoder_helper.h]: ```throw_on_error``` will detect the ```REPLACE_ENFORCE_GLOG``` flag and still throw a exception if we use it directly

**Review by typhoonzero — APPROVED** [review]

File: paddle/fluid/operators/listen_and_serv_op.cc [hunk 2]
```diff
@@ -59,17 +59,16 @@ static void ParallelExecuteBlocks(
...
-          int run_block = idx;  // thread local
-          try {
-            VLOG(3) << "running server block: " << run_block
-                    << "pointer: " << prepared[run_block].get();
-            executor->RunPreparedContext(prepared[run_block].get(), scope);
-          } catch (const std::exception &e) {
-            LOG(ERROR) << "run sub program error " << e.what();
-          }
-        }));
+    fs.push_back(framework::Async([&executor, &prepared, &scope, idx]() {
```
Comment (typhoonzero) [inline paddle/fluid/operators/listen_and_serv_op.cc:62]: seems unused lambda captures in `grpc_client.cc` need to be removed too.

**Pull request merged**